src/app/src/core/services/Utils.js

```diff
@@ -8,10 +8,34 @@ function Utils(Notifications, Storage, $state, $stateParams, ckOptions) {
         $state: $state,
         $stateParams: $stateParams,
         Config: Config,
-        ckOptions: ckOptions
+        ckOptions: ckOptions,
+        /**
+         * Redirect user to previous state
+         * @param {string} defaultStateName default state name
+         */
+        redirectBack: function(defaultStateName) {
+            // Gets a reference to the previous state.
+            var previousState = $previousState.get();
+            // Set default name for the redirect if it is is not specified
+            if (typeof defaultStateName === 'undefined') {
+                defaultStateName = 'home'; // Redirect to home
+            }
+
+            // Wait for CKEeditor instance, so that it can be removed without errors
+            $timeout(function() {
+                // if there is a previousState
+                if (previousState !== null) {
+                    // redirected back to the state we came from
+                    $state.go(previousState.state.name, previousState.params, {reload: true});
+                } else {
+                    // otherwise go to default state
+                    $state.go(defaultStateName, {}, {reload: true});
+                }
+            }, 100);
```

Comment:
Maybe there is some kind of event that we could listen instead trying to ques exact loading time?